### ISSUE TITLE
feat: streaming distinct should keep it append-only

### ISSUE THREAD

**Issue by st1page** [risingwavelabs/risingwave#8118]
```SQL
CREATE SOURCE s(v int);
CREATE MATERIALIZED VIEW AS SELECT DISTINCT ON(k) * from s;
```
Currently, RW's optimizer plan the DISTINCT ON in to a group top-1 executor but it will make append-only property lost. So we want to maintain the append-only of the query.

- [x] https://github.com/risingwavelabs/risingwave/issues/8116
- [ ] add a `LogicalDeduplicate` plan node
  - `LogicalDeduplicate::to_batch` will generate a batch agg with `first_value`
  - `LogicalDeduplicate::to_stream` will generate a materialize executor with `ignore` confliction behavior

**Comment by st1page** [comment]
c.c. @wcy-fdu @xxchan

**Comment by xxchan** [comment]
We can fix its order property together? #7332

**Comment by st1page** [comment]
> We can fix its order property together? https://github.com/risingwavelabs/risingwave/issues/7332

Oh yes, I almost forget it.
